cmd/list.go

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// This file is used to show the list of container checkpoints
+
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/checkpoint-restore/checkpointctl/internal"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+var (
+	defaultCheckpointPath     = "/var/lib/kubelet/checkpoints/"
+	additionalCheckpointPaths []string
+)
+
+func List() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List checkpoints stored in the default and additional paths",
+		RunE:  list,
+	}
+
+	flags := cmd.Flags()
+	flags.StringSliceVarP(
+		&additionalCheckpointPaths,
+		"paths",
+		"p",
+		[]string{},
+		"Specify additional paths to include in checkpoint listing",
+	)
+
+	return cmd
+}
+
+func list(cmd *cobra.Command, args []string) error {
+	allPaths := append([]string{defaultCheckpointPath}, additionalCheckpointPaths...)
+	showTable := false
+
+	table := tablewriter.NewWriter(os.Stdout)
+	header := []string{
+		"Namespace",
+		"Pod",
+		"Container",
+		"Engine",
+		"Time Created",
```

Comment:
We can also add a "Time Checkpointed" field here, thanks to #120

Reply:
Ofcourse we can, but the problem here is that we already have so many columns for the table and it already became too messy while even the terminal is open with full display, should we print it in a different format? 
like - 
```
Checkpoint Name: 
Time Checkpointed:
Time Created:
Namespace:
Pod:
Container:
Engine:
-----------------------------some ending pattern--------------------
```

Reply:
Fair point, what do you think @adrianreber @rst0git

Reply:
> We can also add a "Time Checkpointed" field here

Actually, the current implementation shows the value of `config.CheckpointedAt`. This value indicates the time when the checkpoint was created, and it is also included in the file name.

Example:
```
$ checkpointctl list
Listing checkpoints in path: /var/lib/kubelet/checkpoints/
+-----------+---------------+-----------+--------+---------------------+------------------------------------------------------------------------+
| NAMESPACE |      POD      | CONTAINER | ENGINE |    TIME CREATED     |                            CHECKPOINT NAME                             |
+-----------+---------------+-----------+--------+---------------------+------------------------------------------------------------------------+
| default   | counter-ccpp2 | counter   | cri-o  | 12 Oct 23 11:01 BST | checkpoint-counter-ccpp2_default-counter-2023-10-12T11:01:56+01:00.tar |
+-----------+---------------+-----------+--------+---------------------+------------------------------------------------------------------------+
| default   | counter-ccpp2 | counter   | cri-o  | 12 Oct 23 11:02 BST | checkpoint-counter-ccpp2_default-counter-2023-10-12T11:02:54+01:00.tar |
+-----------+---------------+-----------+--------+---------------------+------------------------------------------------------------------------+
```

`config.CreatedTime` indicates the time when the container was created, i.e., if there are multiple checkpoints of a container, all checkpoints would have the same `CreatedTime` value.